gitops/utils.go

```diff
@@ -32,17 +32,14 @@ func hasDeploymentFinished(objectOld, objectNew client.Object) bool {
 	if oldBinding, ok := objectOld.(*applicationapiv1alpha1.SnapshotEnvironmentBinding); ok {
 		oldCondition = meta.FindStatusCondition(oldBinding.Status.ComponentDeploymentConditions,
 			applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed)
-		if oldCondition == nil {
-			return false
-		}
 	}
 	if newBinding, ok := objectNew.(*applicationapiv1alpha1.SnapshotEnvironmentBinding); ok {
 		newCondition = meta.FindStatusCondition(newBinding.Status.ComponentDeploymentConditions,
 			applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed)
-		if newCondition == nil {
+		if newCondition == nil || newCondition.Status == metav1.ConditionUnknown {
 			return false
 		}
 	}
 
-	return oldCondition.Status == metav1.ConditionUnknown && newCondition.Status != metav1.ConditionUnknown
+	return oldCondition == nil || oldCondition.Status == metav1.ConditionUnknown
```

Comment:
You can reach this part of the code with `objectNew` not being a Binding. Not that is going to happen, but we should fix that.

Reply:
I'd change the code to something like this:
```
if oldObject not a binding -> return
if newObject not a binding -> return

get condition for oldBinding
get condition for newBinding

return (oldCondition == nil || oldCondition.Status == Unknown) && (newCondition != nil && newCondition.Status != Unkown)
```

Reply:
Fixed and added a test case for this